atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolTest.java

```diff
@@ -357,50 +339,44 @@ public void hostIsAutomaticallyAddedOnRefresh() {
         assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
 
         setCassandraServersTo(CASS_SERVER_1, CASS_SERVER_2, CASS_SERVER_3);
-        deterministicExecutor.tick(config.poolRefreshIntervalSeconds(), TimeUnit.SECONDS);
+        deterministicExecutor.tick(POOL_REFRESH_INTERVAL_SECONDS, TimeUnit.SECONDS);
         assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2, CASS_SERVER_3);
     }
 
     @Test
     public void hostsAreNotRemovedOrAddedWhenRefreshIsDisabled() {
-        when(runtimeConfig.servers())
-                .thenReturn(ImmutableDefaultConfig.builder()
-                        .addThriftHosts(CASS_SERVER_1.proxy(), CASS_SERVER_2.proxy())
-                        .build());
+        setupThriftServers(ImmutableSet.of(CASS_SERVER_1.proxy(), CASS_SERVER_2.proxy()));
         when(config.autoRefreshNodes()).thenReturn(false);
 
         setCassandraServersTo(CASS_SERVER_1);
         createClientPool();
         assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
 
         setCassandraServersTo(CASS_SERVER_1, CASS_SERVER_2, CASS_SERVER_3);
-        deterministicExecutor.tick(config.poolRefreshIntervalSeconds(), TimeUnit.SECONDS);
+        deterministicExecutor.tick(POOL_REFRESH_INTERVAL_SECONDS, TimeUnit.SECONDS);
         assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
     }
 
     @Test
     public void hostsAreResetToConfigOnRefreshWhenRefreshIsDisabled() {
-        when(runtimeConfig.servers())
-                .thenReturn(ImmutableDefaultConfig.builder()
-                        .addThriftHosts(CASS_SERVER_1.proxy(), CASS_SERVER_2.proxy())
-                        .build());
+        setupThriftServers(ImmutableSet.of(CASS_SERVER_1.proxy(), CASS_SERVER_2.proxy()));
         when(config.autoRefreshNodes()).thenReturn(false);
 
         setCassandraServersTo(CASS_SERVER_1);
         createClientPool();
         assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
 
-        cassandra.addPool(CASS_SERVER_3);
+        poolServers.add(CASS_SERVER_3);
```

Comment:
this seems like a suspicious change - this is just modifying a hash set instead of testing the Cassandra client pool behaviour, right?

Reply:
`cassandra.addPool` bypasses the actual method: https://github.com/palantir/atlasdb/pull/6214/files#diff-8d42613e8be05141841e2f4f83454300f493eaa51bea43102c964326451e87e2L118-R115

Reply:
Kind of why this is now viewed as an anti-pattern; it makes tests really confusing, as you _think_ you're testing real class behavior, but in reality, you're just interacting with the mock.

Reply:
then is this test even worth including? Is it testing anything other than set behaviour?

Reply:
yeah I'll remove it

Reply:
removed!